colossalai/zero/gemini/chunk/chunk.py

```diff
@@ -339,7 +339,7 @@ def shard_move(self, device: torch.device, force_copy: bool = False):
             if self.cuda_shard:
                 return
 
-            self.cuda_shard = self.cpu_shard.to(get_accelerator().get_current_device())
+            self.cuda_shard = self.cpu_shard.to(get_accelerator().get_current_device(), non_blocking=True)
```

Comment:
Fixed, this `non_blocking` should have been passed in instead of hardcoded

Now this only will be enabled when `enable_async_reduce` is set to `True` and is passed here, `gemini_ddp` will handle the synchronization in `_post_backward`, which was the latest sync point before reduced grads are ready to use